mb/protectli/vault_adl_n: Initial support for VP32XX series

miczyg1 opened the pull request.

miczyg1 committed:
mb/protectli/vault_adl_n: Initial support for VP32XX series

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
3rdparty/dasharo-blobs: Bump for VP32XX blobs

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
.github/workflows/build.yml,build.sh: Add build automation for VP32XX series

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
configs/config.protectli_vp32xx: Bump to rc1

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented:
@SergiiDmytruk something is wrong with the defconfigs checks. It complains on the VP66XX defconfig, that 
```
CONFIG_PCIEXP_L1_SUB_STATE=y
CONFIG_PCIEXP_CLK_PM=y
```

Should be removed. But in the `src/mainboard/protectli/vault_adl_p/Kconfig` has:

```
config PCIEXP_L1_SUB_STATE
	default n

config PCIEXP_CLK_PM
	default n
```

So effectively, the change will disable these features, which is probably not desired...

SergiiDmytruk commented:
Looks fine to me, because `src/soc/intel/common/block/pcie/Kconfig` has:
```
config PCIEXP_CLK_PM
	default y

config PCIEXP_L1_SUB_STATE
	default y
```
as shown in `menuconfig`:
```
...
Defined at src/mainboard/protectli/vault_adl_p/Kconfig:54
  Depends on: VENDOR_PROTECTLI [=y] && BOARD_PROTECTLI_VP66XX [=n]
...
Defined at src/soc/intel/common/block/pcie/Kconfig:16
  Depends on: SOC_INTEL_COMMON [=y] && SOC_INTEL_COMMON_BLOCK [=y] && SOC_INTEL_COMMON_BLOCK_PCIE [=y]
...
```

miczyg1 committed:
mb/protectli/vault_adl_n: Provide fake CLKREQ signals

According to shcematics only Ethernet controllers have CLKREQ signal
routed. However, it is required for each Root Port to have a CLKREQ
signal routed. Otherwise, we hit the issue with MCE caused by lack of
clock, described in fill_fsps_pcie_params. Unfortunately no ADL-N FSP
has PchPcieClockGating and PchPciePowerGating, so one has to fake the
CLKREQ signals for each port. Fortunately the ports with missing CLKREQ
can be assigned to unused CLKREQ signals which are tied to GND per
schematics. This will cause the CLKREQ to be always active for those
devices and not hit the issue with MCE. The issue has been observed by
constant dropping of RDP connection to Windows less than a minute after
logging in.

TEST=RDP connection to Windows is not severed for over 10 minutes.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented:
> Looks fine to me, because `src/soc/intel/common/block/pcie/Kconfig` has:
> 
> ```
> config PCIEXP_CLK_PM
> 	default y
> 
> config PCIEXP_L1_SUB_STATE
> 	default y
> ```
> 
> as shown in `menuconfig`:
> 
> ```
> ...
> Defined at src/mainboard/protectli/vault_adl_p/Kconfig:54
>   Depends on: VENDOR_PROTECTLI [=y] && BOARD_PROTECTLI_VP66XX [=n]
> ...
> Defined at src/soc/intel/common/block/pcie/Kconfig:16
>   Depends on: SOC_INTEL_COMMON [=y] && SOC_INTEL_COMMON_BLOCK [=y] && SOC_INTEL_COMMON_BLOCK_PCIE [=y]
> ...
> ```

It does not look fine to me. Say, I remove those lines from the Kconfig. What will happen?

1. Default values from `src/soc/intel/common/block/pcie/Kconfig`  take priority? OR
2. Default values from `src/mainboard/protectli/vault_adl_p/Kconfig` take priority?

I thought the defaults in mainboard's Kconfig will override the defaults from the rest of Kconfig. But maybe I am mistaken.

miczyg1 committed:
configs/config.protectli_vp66xx: Make CI happy

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

SergiiDmytruk commented:
`src/mainboard/protectli/vault_adl_p/Kconfig` doesn't apply because `BOARD_PROTECTLI_VP66XX` isn't set.

miczyg1 commented:
> There is no order dependence, `src/mainboard/protectli/vault_adl_p/Kconfig` doesn't apply because `BOARD_PROTECTLI_VP66XX` isn't set.

:facepalm: alright... Tha make `olddefconfig` didn't select it previously because vault_adl_p was first in the protectli dir, now it isn't...

miczyg1 committed:
configs/config.protectli_vp32xx: Bump to rc2

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented on src/mainboard/protectli/vault_adl_n/gpio.c:
```diff
@@ -0,0 +1,449 @@
+	/* GPP_H2 - GPIO */
+	PAD_CFG_GPO(GPP_H2, 0, DEEP),
+	PAD_NC(GPP_H3, NONE),
+	PAD_NC(GPP_H4, NONE),
+	PAD_NC(GPP_H5, NONE),
```
NC. What else?

mkopec commented on src/mainboard/protectli/vault_adl_n/Makefile.inc:
```diff
@@ -0,0 +1,11 @@
+ramstage-y += mainboard.c
+
+bootblock-y += die.c
+romstage-y += die.c
```
might also want to add it to ramstage and smm like the other protectli platforms other than adl-p

mkopec commented on src/mainboard/protectli/vault_adl_n/Makefile.inc:
```diff
@@ -0,0 +1,11 @@
+
+ramstage-y += mainboard.c
+
+bootblock-y += die.c
+romstage-y += die.c
```
```suggestion
smm-y += die.c
smm-y += ec.
```

miczyg1 replied:
Not sure if we want to "die" with beeps in SMM.

miczyg1 replied:
174bb97e2aa48b566a2610c73c1b4a201d512d53

miczyg1 replied:
Maybe... Added to all stages, including SMM, anyways.

miczyg1 commented on src/mainboard/protectli/vault_adl_n/board_info.txt:
```diff
@@ -0,0 +1,7 @@
+Category: sbc
+Board URL: TBD
```
Actually the VP2430 will also be a variant here, so not sure how to put all links

miczyg1 replied:
5a6373ffddc9f7be230179dc8bd5aeed334f9570

miczyg1 committed:
mb/protectli/vault_adl_n/board_info.txt: Add URL

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
mb/protectli/vault_adl_n/Makefile.inc: Compile die_notify in all stages

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
mb/protectli/vault_adl_n/gpio.c: Correct GPIOs per schematics

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented on src/mainboard/protectli/vault_adl_n/gpio.c:
```diff
@@ -0,0 +1,449 @@
+	/* GPP_B16 - I2C5_SDA */
+	PAD_CFG_NF(GPP_B16, NONE, DEEP, NF2),
+	/* GPP_B17 - I2C5_SCL */
+	PAD_CFG_NF(GPP_B17, NONE, DEEP, NF2),
```
89bad108d746d4a42e11ab7bb7479a7d8feeb2c3

mkopec reviewed (dismissed):
LGTM, some comments @krystian-hebel remain though

krystian-hebel commented on src/mainboard/protectli/vault_adl_n/bootblock.c:
```diff
@@ -0,0 +1,65 @@
+	u8 pin = (gpio % 10);
+
+	/* There are only 6 configurable sets, 8 pins each */
+	if (gpio < 10 || set > 6 || pin > 7)
```
Then why is there `-1` in the first line of this function? IIUC GPIO numbers start with 10, which would result in `set` equal to 0.

miczyg1 replied:
Ahh right... I missed it.

miczyg1 committed:
mainboard/protectli/vault_adl_n/bootblock.c: Fix GPIO set check

Co-authored-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
configs: Fix EDK2 build with missing subhook module

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
